### PR TITLE
Updates lung code

### LINKED ISSUE #5077: You won't suffocate in any gas as long as it is pressurized
## Reproduction:
be plasmaman
turn off internals while standing in oxygen
see suffocation symbol but don't take damage
probably happens with other species too for example pure nitrogen

you don't suffocate as long as you are breathing atmos that is within the pressure breathing thresholds
your suffocation does not recover though either until you breathe a gas your species can breathe

### LINKED ISSUE #6296: Not requiring to breathe plasma as a Plasmaman.
Issue reported from  Round ID: 35848 (BeeStation Acacia - Roleplay 2)

Reporting client version: 514.1568

This happened to me on mutliple occasions. Sometimes I don't need to breathe plasma even though I changed nothing about myself. One time it's after being revived, another just randomly.

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

Fixes #6296
Fixes #5077

Citadel basically rewrote lung code in the extools -> auxtools switch, this PR just updates us to their latest lung code.

This includes a tested fix for the plasmaman breathing issues. I'm not sure if we care about the other changes or not, but it's probably fine?

## Changelog
:cl:
fix: Fixed plasmamen not actually needing to breathe plasma.
tweak: Updated lung code
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/lungs.dm [hunk 1]
```diff
@@ -82,13 +83,13 @@
 /obj/item/organ/lungs/proc/check_breath(datum/gas_mixture/breath, mob/living/carbon/human/H)
 //TODO: add lung damage = less oxygen gains
 	var/breathModifier = (5-(5*(damage/maxHealth)/2)) //range 2.5 - 5
-	if(H.status_flags & GODMODE)
+	if((H.status_flags & GODMODE))
```
Comment (BrumalLake) [inline code/modules/surgery/organs/lungs.dm]: does this second pair do something special?

Comment (ike709) [reply code/modules/surgery/organs/lungs.dm]: it makes traumatized coders feel safe

Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
	if(H.status_flags & GODMODE)
```

File: code/modules/surgery/organs/lungs.dm [hunk 2]
```diff
@@ -206,15 +204,13 @@
 	for(var/gas in breath.get_gases())
 		if(gas in breath_reagents)
 			var/datum/reagent/R = breath_reagents[gas]
-			H.reagents.add_reagent(R, PP(breath,gas))
+			//H.reagents.add_reagent(R, breath.get_moles(gas) * R.molarity) // See next line
```
Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm:207]: remove commented code

File: code/modules/surgery/organs/lungs.dm [hunk 3]
```diff
@@ -267,51 +262,52 @@
...
-			// In a full miasma atmosphere with 101.34 pKa, about 10 disgust per breath, is pretty low compared to threshholds
-			// Then again, this is a purely hypothetical scenario and hardly reachable
-			owner.adjust_disgust(0.1 * miasma_pp)
-
-			breath.adjust_moles(GAS_MIASMA, -gas_breathed)
+			if(miasma_pp > MINIMUM_MOLES_DELTA_TO_MOVE)
+
+				//Miasma sickness
+				if(prob(0.05 * miasma_pp))
+					var/datum/disease/advance/miasma_disease = new /datum/disease/advance/random(TRUE, 2,3)
```
Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
					var/datum/disease/advance/miasma_disease = new /datum/disease/advance/random(TRUE, 2, 3)
```

File: code/modules/surgery/organs/lungs.dm [hunk 4]
```diff
@@ -267,51 +262,52 @@
...
+					miasma_disease.try_infect(owner)
+
+				// Miasma side effects
+				switch(miasma_pp)
+					if(1 to 5)
+						// At lower pp, give out a little warning
+						SEND_SIGNAL(owner, COMSIG_CLEAR_MOOD_EVENT, "smell")
+						if(prob(5))
+							to_chat(owner, "<span class='notice'>There is an unpleasant smell in the air.</span>")
+					if(5 to 15)
```
Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
					if(6 to 15)
```

File: code/modules/surgery/organs/lungs.dm [hunk 5]
```diff
@@ -267,51 +262,52 @@
...
+						//At somewhat higher pp, warning becomes more obvious
+						if(prob(15))
+							to_chat(owner, "<span class='warning'>You smell something horribly decayed inside this room.</span>")
+							SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "smell", /datum/mood_event/disgust/bad_smell)
+					if(15 to 30)
```
Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
					if(16 to 30)
```

File: code/modules/surgery/organs/lungs.dm [hunk 6]
```diff
@@ -267,51 +262,52 @@
...
+						//Small chance to vomit. By now, people have internals on anyway
+						if(prob(5))
+							to_chat(owner, "<span class='warning'>The stench of rotting carcasses is unbearable!</span>")
+							SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "smell", /datum/mood_event/disgust/nauseating_stench)
+							owner.vomit()
+					if(30 to INFINITY)
```
Comment (Archanial) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
					if(31 to INFINITY)
```

File: code/modules/surgery/organs/lungs.dm (hunk 2 above)
Comment (ike709) [reply code/modules/surgery/organs/lungs.dm:207]: No, it's relevant.

**Review by Archanial — APPROVED** [review]

**Pull request merged**